### DISCUSSION TITLE
Anycubic M3, elephant foot calibration tool

### DISCUSSION THREAD

**Discussion by alwayzbelearnin** [sn4k3/UVtools#528] [General]
Printer: Anycubic M3
UVTools: 3.5.5
Slicer: Lychee, 8x AA, 20% grey

I'm attempting to combat elephant foot when printing directly on build plate. I dived into the the subject over the last few days and it seems like Anycubic firmware in general is not very friendly to tinkering. I was hopeful UVTools would be able to solve my problem with the elephant foot calibration tool. I used both the morph-erode and wall-dimming settings with an AA file. I expected the erode to show some difference in the print, and was hopeful for the wall-dimming to work. I have a successful looking preview (seen below), however all the test pieces look identical.  My calibration pics are below as well.

I was hoping for some insight into this, or to hear of others' successes with the M3 series printers. 

Is there anything else I should alter in Lychee regarding the AA or grey percentage? Is there anything I should try within UVTools? Is the anycubic firmware able to process the UVTools tool?

<img width="1264" alt="Screen Shot 2022-07-29 at 10 42 12 PM" src="https://user-images.githubusercontent.com/110271193/181867117-5de435c5-ce2a-4ef8-a402-2467cbd6ae4c.png">
<img width="817" alt="Screen Shot 2022-07-29 at 10 49 53 PM" src="https://user-images.githubusercontent.com/110271193/181867569-4ec80c8f-c4ea-4e05-8538-cc90b067a778.png">
<img width="901" alt="Screen Shot 2022-07-29 at 10 50 07 PM" src="https://user-images.githubusercontent.com/110271193/181867573-23871b42-74ee-41f5-b88b-8fd56f8fdec0.png">
<img width="896" alt="Screen Shot 2022-07-29 at 10 50 22 PM" src="https://user-images.githubusercontent.com/110271193/181867582-a3b6d851-82a5-4e34-a42b-c58f7c5ade9f.png">

**Comment by sn4k3** [comment]
Have you read https://github.com/sn4k3/UVtools/wiki/Calibration:-Elephant-Foot ?

AA for anycubic machines don't work so great, it's fake AA.

> Is there anything else I should alter in Lychee regarding the AA or grey percentage? 

This is to be applied in UVtools, so you should not change slicer settings after you come to the optimal result.

Now you can run the test as shown and observe the results

**Reply by alwayzbelearnin** [reply]
Yes I did read the wiki, thanks for having that there. I already ran the above print and there were no differences in any of the models (neither with erode nor wall dimming). Does the UVTool wall dimming work with anycubic firmware (will the brightness differences in the test prints be executed/read by my printer)? I also don't understand why the erode didn't show any difference, since it is removing pixels from the print file?

**Reply by sn4k3** [reply]
> were no differences in any of the models

Not even one small difference?

> Does the UVTool wall dimming work with anycubic firmware

I hear that AA is broken on new models, i'm not sure. But it should work... But it's fake AA, which means, fractions of exposure time instead of faded pixels. Also you must have AA enabled on your test file (On slicer you must choose AA and opt for the highest count) otherwise it won't be used.
**For anycubic i don't recommend the wall dimming method.**

> I also don't understand why the erode didn't show any difference

The smaller the lcd pixel and more bottom exposure time the more iterations you have to run.
Some people have to run from 10px to 20px. 
Continue with test up to desired result, don't be afraid to push more iterations.
Go just with erode and continue from 7px to 17px

While you can't use the newest method to get rid of elephant foot (Due poor anycubic firmware), also read this: https://blog.honzamrazek.cz/2022/01/prints-not-sticking-to-the-build-plate-layer-separation-rough-surface-on-a-resin-printer-resin-viscosity-the-common-denominator to know more

**Reply by alwayzbelearnin** [reply]
> Not even one small difference?

No perceivable differences. Here is a pic of original unmodified with highest erode and 120 vs 240 brightness.
![IMG_0364](https://user-images.githubusercontent.com/110271193/181915476-6cbee077-24f5-4fd0-9ba0-78bce5380260.jpg)

 >I hear that AA is broken on new models, i'm not sure. But it should work... But it's fake AA, which means, fractions of exposure time instead of faded pixels. Also you must have AA enabled on your test file (On slicer you must choose AA and opt for the highest count) otherwise it won't be used.
For anycubic i don't recommend the wall dimming method.

Should I try the greyscale calibration tool to test it out? Does it use the same algorithm as your pixel dimming tool? I can try and dink around with different AA settings in my slicer before bringing into UVTools too to see if there is any difference.

> The smaller the lcd pixel and more bottom exposure time the more iterations you have to run.
Some people have to run from 10px to 20px.
Continue with test up to desired result, don't be afraid to push more iterations.
Go just with erode and continue from 7px to 17px

I'll try this, thanks for the tip.

I've read a few of Jan's posts, he actually set me on the path to improving my elephant foot and onto UVTools. Unfortunately my damn anycubic firmware won't allow per layer settings and rest time after retract, etc. So annoying.

**Reply by sn4k3** [reply]
There is a subtle difference between 1 and 7. You need to push more iterations.

> Should I try the greyscale calibration tool to test it out? 

You can try it out. You will notice that some grey levels are in same level because thier fake AA works with thresholds.

> Does it use the same algorithm as your pixel dimming tool?

Yes.

> anycubic firmware

One of the worse! They can have the best firmware in a day of codding, still they don't care to make thier machines better with virtually no cost!

**Reply by alwayzbelearnin** [reply]
No luck with any of my test prints with the elephant foot calibration tool. I tried multiple sets with AA of 2x and 8x at various levels of grey coming from my slicer for the dimming pixel tool (with wall thickness of 20 and 40 px, brightness from 40-240) and erode tools (2-20 pixel by 2 step) . No consistent or discernible difference in any of them. Frustrated with anycubic to say the least. 

Also printed the grey calibration tool shown below. Not sure that it helps my elephant foot cause at this point, but at least I got some information out of it. Really below 165 (2.36 sec) would be a waste to include in my AA files. 

At this point, may just try to not use AA and print everything on rafts with supports unfortunately. 

Printed 145-255 brightness (I converted to seconds like I though that was a good idea...) in steps of 10.
![IMG_0403](https://user-images.githubusercontent.com/110271193/182053153-ded3f3d0-342d-4db0-abaa-e5963ffa9d71.jpg)
![IMG_0404](https://user-images.githubusercontent.com/110271193/182053159-091e9b42-aed1-4a04-8bd7-cb8d12201264.jpg)

**Reply by sn4k3** [reply]
The erode with 20px didn't make a difference? :O If so something is wrong.

You grey pie test show that at least AA is working, and as expected 4 sections exposed the same since your machine use exposure time threshold, which means with AAx4 you can have only 4 exposure times and all in between will expouse with all same time, with AAx8 you can have 8 different times and so on...

## thresholds:
aa 1:  127
aa 2:  255 127
aa 4:  255 191 127 63
aa 8:  255 223 191 159 127 95 63 31

**Reply by alwayzbelearnin** [reply]
Yeah not sure why the erode didn't work. I might just try the pixel arithmetic on a sliced model of mine to see how it works outside the calibration tool. 

Thanks for your help and timely replies. UVTools is an awesome tool, I wish I could fully utilize its potential with my printer. Next resin printer I will make sure plays nice. lol

**Reply by sn4k3** [reply]
Erode must work... Can you send me the file you print, attach here the file from pen drive which you selected to print.

**Comment by alwayzbelearnin** [comment]
I printed all of these. I think the APAA2 I didn’t do the erode on though.

Craig

From: Tiago Conceição ***@***.***>
Date: Monday, August 1, 2022 at 11:09 AM
To: sn4k3/UVtools ***@***.***>
Cc: alwayzbelearnin ***@***.***>, Author ***@***.***>
Subject: Re: [sn4k3/UVtools] Anycubic M3, elephant foot calibration tool (Discussion #528)

Erode must work... Can you send me the file you print, attach here the file from pen drive which you selected to print.

**Reply by sn4k3** [reply]
You cant attach files via email. Must be there or external link

**Reply by alwayzbelearnin** [reply]
Shoot sorry, GitHub and discord are both very new to me.
[EF compensation.zip](https://github.com/sn4k3/UVtools/files/9235395/EF.compensation.zip)

**Reply by sn4k3** [reply]
Any of that files should have produce a visible change. If resin is not transparent.

**Reply by alwayzbelearnin** [reply]
Yea used opaque resin from a new bottle. Pretty ideal operating conditions. No sure why or how my M3's firmware could overcome the processing from UVTools. Have you heard of anybody using one of the M3's with any luck?

**Reply by sn4k3** [reply]
If you are using transparent resin you have the problem of overcuring the previous layer. Try to increase bottom layers to 20/30

**Reply by alwayzbelearnin** [reply]
I didn’t use transparent resin. It was sky blue solid.

**Reply by sn4k3** [reply]
Then i'm out of ideias...
I would do a dry run with a papper and see what LCD is exposure just to be sure that is displaying erroded models just right
